### PR TITLE
More documentation

### PR DESCRIPTION
**Pull request by alongreyber**
Added high level documentation overviews for execution and planning. These are good for onboarding onto the project. Couldn't quite finish the planning documentation but got pretty close to done.

### CODE REVIEW

File: docs/PLANNING_OVERVIEW.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+```mermaid
+graph TD
+        n0(( category biolink:MolecularEntity ))
+        n1(( id MONDO:0005148 ))
+        n2(( category biolink:GeneOrGeneProduct ))
+        n1-- biolink:treated_by<br/>kp3 = Disease-treated_by->MolecularEntity -->n0
+        n0-- biolink:affects_abundance_of<br/>kp2 = MolecularEntity-decreases_abundance_of->GeneOrGeneProduct -->n2
+```
+
+TODO finish
```
Comment (davefol) [inline docs/PLANNING_OVERVIEW.md:181]: What else needs to be said here?

**Review by davefol — CHANGES_REQUESTED** [review]

File: docs/COMPONENTS.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+* `normalizer.py` defines a Python client for the node normalizer service: https://github.com/TranslatorSRI/NodeNormalization, https://nodenormalization-sri.renci.org/docs
+* `profiler.py` handles request profiler
+* `query_planner.py` contains tools for planning query graph traversals
+* `results.py` **probably obsolete**
+* `scoring.py` **probably obsolete**
+* `server.py` builds the [FastAPI](https://fastapi.tiangolo.com/) server and endpoints
+* `storage.py` defines interfaces for accessing and manipulating Redis storage
+* `trapi_openapi.py` defines the TRAPI subclass of FastAPI to add the common TRAPI elements to the OpenAPI schema
+* `trapi.py` defines utilities for TRAPI messages, including normalizing and merging
+* `traversal.py` **probably obsolete**
```
[the remark this answers is not in the document]
Comment (alongreyber) [reply docs/COMPONENTS.md:33]: I *think* this actually does something because get_traversals will raise NoAnswersError if it can't find a traversal.

**Comment by alongreyber** [comment]
This PR looks good except that the `docs/PLANNING_OVERVIEW.md` document is very out of date. I don't really have a good grasp on how the new query planning works. @uhbrar would you be able to update that document? Patrick wrote the docs/QUERY_PLANNING.md document but that isn't really an overview of how it works–it's specifically focused on some statistical optimization stuff.  I think that a planning overview for people new to the code would be helpful to have.

If you also don't know how it works I can dive in and learn about it but I thought I would ask.

**Comment by alongreyber** [comment]
@davefol I made a milestone for cleaning up dead code so feel free to create issues and link them to it

**Comment by davefol** [comment]
> This PR looks good except that the `docs/PLANNING_OVERVIEW.md` document is very out of date. I don't really have a good grasp on how the new query planning works. @uhbrar would you be able to update that document? Patrick wrote the docs/QUERY_PLANNING.md document but that isn't really an overview of how it works–it's specifically focused on some statistical optimization stuff. I think that a planning overview for people new to the code would be helpful to have.
> 
> If you also don't know how it works I can dive in and learn about it but I thought I would ask.

perfect is the enemy of good yada yada. I'll make an issue for updaating PLANNING_OVERVIEW.md and tentatively assign @uhbrar to it. I don't want it to block this PR because this is a lot of really good stuff.

**Comment by kennethmorton** [comment]
I think I might have the clearest (remaining) picture of the query planning algorithm. I'll put this on my list for documentation.

From my current understanding the algorithm we have is pretty solid, and shouldn't need to be overhauled for a while, but we might want to add a few bells and whistles. However those bells and whistles don't offer as much as optimizations elsewhere right now.

**Comment by davefol** [comment]
I'll add @kennethmorton to the assigned members for issue #321

**Comment by uhbrar** [comment]
I'll take a look at it as well, although I don't actually know how it works precisely, except for some parts that Patrick shared with me before he left. That being said, besides Kenny, I might have the best chance of getting it fully figured out, and it'll be a good opportunity to look through the code and learn about it for me anyway.

**Review by davefol — APPROVED** [review]

**Pull request merged**